src/firefly/js/ui/UploadTableChooser.js

```diff
@@ -127,9 +127,37 @@ function existingTableSubmit(request,setUploadInfo) {
     return false;
 }
 
-function getSelectedColumns(columns) {
-    const {lonCol='', latCol=''} = findTableCenterColumns({tableData:{columns}}) ?? {}; //centerCols
-    const columnsSelected = columns.map((col) => col.name === lonCol || col.name === latCol? ({...col, use:true}) :  ({...col, use:false})); //select position cols only
+/*
+    if showUploadTableChooser is called with a defaultColsEnabled object, use the colTypes and colCount to set the selected columns
+    for this uploaded table
+ */
+function defaultColumnsSelector(columns,colTypes,colCount) {
+    let cols = [];
+    if  (columns == null) return null;
+    let count = 0;
+    for (let i=0; i< columns?.length; i++) {
+        if (count < colCount && colTypes.includes(columns[i].type)) {
+            cols.push(columns[i]);
+            count ++;
+        }
+    }
+    cols = cols.map((col) => col.name.replace(/^"(.*)"$/, '$1'));
```

Comment:
Not 100% what you want here, I am assuming you want the first 2 column with the given `colTypes`.  There are existing util function in TableUtil.js you can use.  `getColsByType` and `stripColumnNameQuotes`.  These functions includes logic that you missed, like hidden column and column names with double-quotes in them.